components/Chat/ChatInput.tsx

```diff
@@ -1,51 +1,89 @@
 import { TvMinimalPlay } from "lucide-react";
 import { usePathname } from "next/navigation";
+import { useState, useEffect, useRef } from "react";
 import useProfileSearch from "@/hooks/useProfileSearch";
 import getZoraPfpLink from "@/lib/zora/getZoraPfpLink";
 import { useChatProvider } from "@/providers/ChatProvider";
+import MediaSelect from "../MediaSelect";
 import SubmitButton from "./SubmitButton";
 import Suggestions from "./Suggestions";
 import manifest from "@/public/manifest.json";
-
 const ChatInput = () => {
   const {
     handleSubmit,
     handleInputChange,
     input,
     pending,
+    messages,
+    append,
   } = useChatProvider();
   const { profile } = useProfileSearch();
+  const [showMediaSelect, setShowMediaSelect] = useState(false);
   const color = (!pending && input.length > 0) ? "#000000" : manifest.theme_color;
-
+  const lastProcessedMessageRef = useRef<string>("");
   const pathname = usePathname();
-
   const isNewChat = pathname === "/";
-
+  useEffect(() => {
+    if (messages.length >= 2) {
+      const lastMessage = messages[messages.length - 1];
+      const secondLastMessage = messages[messages.length - 2];
+      const messageKey = `${lastMessage.id}-${secondLastMessage.id}`;
+      if (messageKey === lastProcessedMessageRef.current) {
+        return;
+      }
+      if (lastMessage.role === "assistant" &&
+          secondLastMessage.role === "user" &&
+          secondLastMessage.content.toLowerCase().includes("create")) {
```

Comment:
Incorrect implementation
- actual: `hard-code` checking if message includes `create` text
- required: checking `tool` from LLM response to determine UI components to render.